lib/restrictions.js

```diff
@@ -17,7 +19,10 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
   await promisify(database.createIndexes)([{
     // for getting restrictions for a particular resource within a zone
     collection: 'resource-restriction-restriction',
-    fields: {'restriction.resource': 1, 'restriction.zone': 1},
+    fields: {
+      'restriction.resource': 1,
+      'restriction.zone': 1,
+      'restriction.methodOptions': 1},
```

Comment:
Yes, this will be a breaking change.

I have added validation for `restriction` so it will always contain `methodOptions` and subsequently `duration`. Records that share `resource` and `zone` but have a different `duration` are now not counted as duplicates, and I do not think that is a new feature, rather how this was intended to be implemented but not written yet.

At this time, to the best of my knowledge this does not affect any of the previous data in `bedrock-resource-restriction` or any of the projects that depend on it, because the data should already be in the exact format that is now being mandated by the validator.

Reply:
I'm going to mark this as resolved primarily because it's referencing a now obsolete line of code.  I'm adding it to the office hours agenda for tomorrow to ensure we get additional eyes on the index change.